sql/_05_plcsql/_01_testspec/_02_declaration/_09_percent_type/cases/31_bugfix_percent_type.sql

```diff
@@ -0,0 +1,55 @@
+--+ server-message on
+
+-- Verification for %type bugfix.
+
+
+-- CBRD-25172 : %TYPE in SP parameter and return types
+create or replace function test_percent_type_in_sig(c dual.dummy%TYPE) return int as
+begin
+    return null;
+end;
+
+select test_percent_type_in_sig('a');
+
+create or replace function test_percent_type_in_sig(c int) return dual.dummy%TYPE as
+begin
+    return null;
+end;
+
+select test_percent_type_in_sig(12);
+
+drop function test_percent_type_in_sig;
+
+
+-- CBRD-25174 : Resolved types of %TYPEs must be used in type check and code generation stage
+create or replace procedure test_percent_typed_arg_to_out_param as
+    c dual.dummy%type;
+
+    procedure poo(c out char(1)) as
+    begin
+        dbms_output.put_line('Call OK');
+    end;
+begin
+    poo(c);
+end;
+call test_percent_typed_arg_to_out_param();
+
+drop procedure test_percent_typed_arg_to_out_param;
+

```

Comment:
아래의 in parameter의 검증을 추가할 것을 제안드립니다.
예상 결과는
print in value: z
c value: z
입니다.

```suggestion

create or replace procedure test_percent_typed_arg_to_in_param() as
    c dual.dummy%type :='z' ;
    procedure poo(in_param in char(1)) as
    begin
        dbms_output.put_line('print in value: ' + in_param);
    end;
begin
    poo(c);
    dbms_output.put_line('c value: ' + c);
end;
call test_percent_typed_arg_to_in_param();
drop procedure test_percent_typed_arg_to_in_param;

```

Reply:
추가하였습니다.